### PR TITLE
feat: non-trigger script and dynamic collider

### PR DESCRIPTION
**Pull request by yangfengzzz**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/core/src/physics/DynamicCollider.ts [hunk 1]
```diff
@@ -2,24 +2,215 @@ import { IDynamicCollider } from "@oasis-engine/design";
...
+
+  set freezeRotation(newValue: boolean) {
+    this._freezeRotation = newValue;
+    (<IDynamicCollider>this._nativeCollider).setFreezeRotation(newValue);
+  }
+
+  /**
+   * The particular rigid dynamic lock flag.
+   */
+  get constraints(): number {
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:194]: Use `DynamicColliderConstraints` as type is better than number

File: packages/core/src/physics/DynamicCollider.ts [hunk 2]
```diff
@@ -59,3 +273,35 @@ export class DynamicCollider extends Collider {
...
+  ContinuousDynamic,
+  /// Speculative continuous collision detection is on for static and dynamic geometries
+  ContinuousSpeculative
+}
+
+/**
+ * Use these flags to constrain motion of dynamic collider.
+ */
+export enum DynamicColliderConstraints {
+  /** Freeze motion along the X-axis. */
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:295]: You can add `None=0`, We can constrain properties entirely using enum operations,not using number any more.

File: packages/core/src/physics/DynamicCollider.ts [hunk 3]
```diff
@@ -59,3 +273,35 @@ export class DynamicCollider extends Collider {
     this._updateFlag.flag = false;
   }
 }
+
+/**
+ * The collision detection mode constants.
+ */
+export enum CollisionDetectionMode {
+  /// Continuous collision detection is off for this dynamic collider.
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:281]: Write comments use `/**/` not `///`

File: packages/core/src/physics/DynamicCollider.ts [hunk 4]
```diff
@@ -46,6 +237,29 @@ export class DynamicCollider extends Collider {
     (<IDynamicCollider>this._nativeCollider).addTorque(torque);
   }
 
+  /**
+   * Moves kinematically controlled dynamic actors through the game world.
+   * @param position The desired position for the kinematic actor
+   * @param rotation The desired rotation for the kinematic actor
+   */
+  setKinematicTarget(position: Vector3, rotation: Quaternion) {
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:245]: It is more difficult to use for the case where only the target is set and the rotation is not considered.
Need for a fuller discussion.

File: packages/core/src/Script.ts [hunk 5]
```diff
@@ -86,6 +86,25 @@ export class Script extends Component {
...
+   * @remarks onTriggerStay is called every frame while the collision stay.
+   * @param other ColliderShape
+   */
+  onContactExit(other: ColliderShape): void {}
+
+  /**
+   * Called when the collision exit.
+   * @param other ColliderShape
+   */
+  onContactStay(other: ColliderShape): void {}
```
[the remark this answers is not in the document]
Comment (yangfengzzz) [reply packages/core/src/Script.ts]: +1

File: packages/core/src/physics/DynamicCollider.ts [hunk 6]
```diff
@@ -46,6 +237,29 @@ export class DynamicCollider extends Collider {
...
+    (<IDynamicCollider>this._nativeCollider).setKinematicTarget(position, rotation);
+  }
+
+  /**
+   * Forces a collider to sleep at least one frame.
+   */
+  putToSleep() {
```
[the remark this answers is not in the document]
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts]: +1

File: packages/core/src/physics/DynamicCollider.ts (hunk 1 above)
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts]: +1

File: packages/core/src/physics/DynamicCollider.ts (hunk 2 above)
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts:297]: +1

File: packages/core/src/physics/DynamicCollider.ts [hunk 7]
```diff
@@ -2,24 +2,215 @@ import { IDynamicCollider } from "@oasis-engine/design";
...
+   * Controls whether physics will change the rotation of the object.
+   */
+  get freezeRotation(): boolean {
+    return this._freezeRotation;
+  }
+
+  set freezeRotation(newValue: boolean) {
+    this._freezeRotation = newValue;
+    (<IDynamicCollider>this._nativeCollider).setFreezeRotation(newValue);
+  }
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:189]: Maybe the function is similar to constraints

File: packages/core/src/physics/PhysicsManager.ts [hunk 8]
```diff
@@ -62,6 +102,18 @@ export class PhysicsManager {
     }
   };
 
+  get gravity(): Vector3 {
+    return this._gravity;
+  }
+
+  set gravity(g: Vector3) {
```
Comment (GuoLei1990) [inline packages/core/src/physics/PhysicsManager.ts:109]: use `value` insteadof `g`

Comment (GuoLei1990) [reply packages/core/src/physics/PhysicsManager.ts:109]:
```ts
set gravity(value: Vector3) {
    const gravity = this._gravity;
    if (gravity !== value) {
      value.cloneTo(gravity);
    }
    this._nativePhysicsManager.setGravity(gravity);
  }
```

File: packages/core/src/physics/PhysicsManager.ts [hunk 9]
```diff
@@ -12,11 +12,51 @@ export class PhysicsManager {
   /** @internal */
   static _nativePhysics: IPhysics;
 
+  private _gravity: Vector3 = new Vector3();
   private _nativePhysicsManager: IPhysicsManager;
   private _physicalObjectsMap: Record<number, ColliderShape> = {};
-  private _onContactEnter = (obj1: number, obj2: number) => {};
-  private _onContactExit = (obj1: number, obj2: number) => {};
-  private _onContactStay = (obj1: number, obj2: number) => {};
+  private _onContactEnter = (obj1: number, obj2: number) => {
```
Comment (GuoLei1990) [inline packages/core/src/physics/PhysicsManager.ts:18]: Try best not use `()=>`, maybe can get better performance

File: packages/core/src/physics/DynamicCollider.ts [hunk 10]
```diff
@@ -46,11 +237,34 @@ export class DynamicCollider extends Collider {
     (<IDynamicCollider>this._nativeCollider).addTorque(torque);
   }
 
+  /**
+   * Moves kinematically controlled dynamic actors through the game world.
+   * @param position The desired position for the kinematic actor
+   * @param rotation The desired rotation for the kinematic actor
+   */
+  setKinematicTarget(position: Vector3, rotation: Quaternion): void {
```
Comment (GuoLei1990) [inline packages/core/src/physics/DynamicCollider.ts:245]: Use `move` insteadof `setKinematicTarget ` and add function overload:

-  `move(position: Vector3, rotation: Quaternion)`
-  `move(position: Vector3)`
-  `move(rotation: Quaternion)`

Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts]: +1

File: packages/core/src/physics/PhysicsManager.ts (hunk 8 above)
Comment (yangfengzzz) [reply packages/core/src/physics/PhysicsManager.ts]: +1

File: packages/core/src/physics/DynamicCollider.ts (hunk 7 above)
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts:177]: +1

File: packages/core/src/physics/DynamicCollider.ts [hunk 11]
```diff
@@ -2,24 +2,215 @@ import { IDynamicCollider } from "@oasis-engine/design";
...
+  }
+
+  /**
+   * The angular damping of the dynamic collider.
+   */
+  get angularDamping(): number {
+    return this._angularDamping;
+  }
+
+  set angularDamping(newValue: number) {
```
[the remark this answers is not in the document]
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts]: +1

File: packages/core/src/physics/DynamicCollider.ts (hunk 4 above)
Comment (yangfengzzz) [reply packages/core/src/physics/DynamicCollider.ts]: +1

**Review by GuoLei1990 — APPROVED** [review]
+1

File: packages/core/src/physics/PhysicsManager.ts [hunk 12]
```diff
@@ -62,6 +102,18 @@ export class PhysicsManager {
...
+  }
+
+  set gravity(value: Vector3) {
+    const gravity = this._gravity;
+    if (this._gravity !== value) {
+      value.cloneTo(gravity);
+    }
+    this._nativePhysicsManager.setGravity(gravity);
+  }
+
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/physics/PhysicsManager.ts:116]: if (gravity !== value) {

}

Comment (yangfengzzz) [reply packages/core/src/physics/PhysicsManager.ts:116]: +1

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**